src/shuffler-codereview.c

```diff
@@ -0,0 +1,794 @@
+    AdjustedObjects[objNum].t.introText = tt->introText;
+    AdjustedObjects[objNum].t.postbattleText = tt->postbattleText;
+    AdjustedObjects[objNum].t.name = tt->trainerName;
+    AdjustedTemplates[objNum].graphicsId = tt->graphicsId;
+    AdjustedTemplates[objNum].flagId = ShuffledFlagNumberByObjectEventId(objNum + 1);
+
+    if (partnerObjNum != 0xFF && (AdjustedObjectsDirty & (1 << partnerObjNum)) == 0) {
+        // combine the parties
+        int partySizeA = AdjustedObjects[objNum].t.trainer.partySize >> 1;
+        if (partySizeA < 1) {
+            partySizeA = 1;
+        } else if (partySizeA > 3) {
+            partySizeA = 3;
+        }
+        int partySizeB = AdjustedObjects[partnerObjNum].t.trainer.partySize >> 1;
+        if (partySizeB < 1) {
+            partySizeB = 1;
+        } else if (partySizeB > 3) {
+            partySizeB = 3;
+        }
+
+        int copyDestIndex;
+        if (partySizeA == 3) {
+            if (partySizeB == 1) {
+                copyDestIndex = 3;
+            } else {
+                copyDestIndex = 4;
+            }
+            memcpy(&AdjustedObjects[objNum].t.party[copyDestIndex], &AdjustedObjects[objNum].t.party[2], sizeof(struct TrainerMonItemCustomMoves));
+        }
+        
+        if (partySizeA >= 2) {
+            memcpy(&AdjustedObjects[objNum].t.party[2], &AdjustedObjects[objNum].t.party[1], sizeof(struct TrainerMonItemCustomMoves));
+        }
+
+        if (partySizeB == 3) {
+            memcpy(&AdjustedObjects[objNum].t.party[partySizeA + 2], &AdjustedObjects[partnerObjNum].t.party[2], sizeof(struct TrainerMonItemCustomMoves));
+        }
+
+        if (partySizeB >= 2) {
+            if (partySizeA == 1) {
+                copyDestIndex = 2;
+            } else {
+                copyDestIndex = 3;
+            }
+            memcpy(&AdjustedObjects[objNum].t.party[copyDestIndex], &AdjustedObjects[partnerObjNum].t.party[1], sizeof(struct TrainerMonItemCustomMoves));
+        }
+
+        memcpy(&AdjustedObjects[objNum].t.party[1], &AdjustedObjects[partnerObjNum].t.party[0], sizeof(struct TrainerMonItemCustomMoves));
+
+        // copy relevant data
+        AdjustedObjects[partnerObjNum].t.trainer.party.ItemCustomMoves = AdjustedObjects[objNum].t.party;
+
+        AdjustedObjects[objNum].t.trainer.partySize = partySizeA + partySizeB;
+        AdjustedObjects[partnerObjNum].t.trainer.partySize = partySizeA + partySizeB;
+
+        AdjustedObjects[objNum].t.trainer.doubleBattle = TRUE;
+        AdjustedObjects[partnerObjNum].t.trainer.doubleBattle = TRUE;
+
+        RedirectedTrainerFlags |= 1 << partnerObjNum;
+        MYLOG("DeclareTrainer RedirectedTrainerFlags: %d", RedirectedTrainerFlags);
+        AdjustedTemplates[partnerObjNum].flagId = AdjustedTemplates[objNum].flagId;
+    }
+}
+
+void DeclareWildMon(u8 objNum, u8 partnerObjNum) {
+    MirrorMapData();
+    AdjustedObjectsDirty ^= (1 << objNum);
+    int i;
+    if (WILD_LEGEND_ODDS == 0) {
```

Comment:
354-367 would be much easier to parse as:

if (WILD_LEGEND_ODDS != 0 && (itinymt32_generate_uint32(&currentRoomSeed) % WILD_LEGEND_ODDS) == 0) {
species = possibleWildLegends[tinymt32_generate_uint32(&currentRoomSeed) % POSSIBLE_WILD_MON_LEGENDARY];
} else {
species = possibleWildMon[tinymt32_generate_uint32(&currentRoomSeed) % POSSIBLE_WILD_MON];
}